articles/introduction-to-room-db/index.md

```diff
@@ -0,0 +1,155 @@
+
+With that done, let's now proceed to the main task.
+
+### Step 2 — Creating the Entity
+Create a package named `db` and add a Kotlin class inside it. As mentioned earlier, this is the class that we will use to define our table, i.e, its name and column properties.
+
+Add the following code to the class.
+```Kotlin
+@Entity
+data class ToDo( @PrimaryKey val content: String)
+```
+Since we are going to record a sentence for each to-do item, we only need one field in the table. We annotate the field with `@PrimaryKey` to declare it the primary key for our table. There are many annotations for table fields. You can read more about the fields [here](https://developer.android.com/training/data-storage/room/defining-data)
+
+### Step 3 — Creating the DAO
+Now that we have defined our table, let's go ahead and create functions to query our table. Still, in the `db` package, create another file and name it `DAO`. We need to create three functions. One to get the ToDO list, another to save a ToDo item, and another one to delete a ToDo item.
+
+Add the code below to implement this.
+```Kotlin
+@Dao
+interface Dao {
+    @Insert
+    fun addItem(todo: ToDo)
+
+    @Query("SELECT * FROM ToDo")
+    suspend fun getList()
+
+    @Query("DELETE FROM ToDo WHERE content = :content")
+    suspend fun deleteItem(content: String)
+}
+```
+The `DAO` interface has three major annotations used to query the database.
+ 1. `@Insert` - This is used to insert data into a table. The item passed in the function that is annotated with this function is added to the database.
+ 2. `@Query` - This is used when one wants to write an SQL statement. It receives an SQL statement as its parameter. We have used this annotation to get all the ToDo items and to delete one item. We use `@query` to delete an item since there is no particular annotation to delete a single item.
+ 3. `@Delete` - This annotation is used to delete the contents of a database.
+
+We have marked the two functions as suspending so that they do not block the UI.
+
+That's all we need for the DAO. Let's now create the database class.
+
+### Step 4 — Creating the Database class
+As mentioned earlier, the database class uses the data access object to issue queries to the database. For this reason, the class must hold a reference to the DAO. Write the code below in the database class
+
+```Kotlin
+@Database(entities = [Note::class], version = 1, exportSchema = false)
+abstract class NoteDatabase : RoomDatabase(){
+    abstract fun noteDao(): NoteDao
+
+    companion object {
+        @Volatile
+        private var INSTANCE: NoteDatabase? = null
+
+        fun getDatabase(context: Context): NoteDatabase {
+            return INSTANCE ?: synchronized(this) {
```

Comment:
I think there is a lot more going on here that you have not explained. Could you provide info on what instance we are getting, why we are getting it, and how...I think that will cover the entire snippet. And about the parameters passed in the `@Database` annotation.

This is a technical issue